doc/ref_model/chapters/chapter03.md

```diff
@@ -543,33 +543,52 @@ SyncE architecture minimally requires replacement of the internal clock of the E
-The storage model provides a relatively simple way for the storage consumer to specify / select their storage needs. This is shown in the follow table which highlights value of the storage classes and the epic use case for usage pattern.
+* (Comparitive) Cost - is monetory value / unit of end user storage capacity (see below for indicative comparative costs)
+* Performance - is defined by IOPS / Latency / Throught Put as typically each of these increases with succesive genarations of storage
+* Capacity - consumption needs are represented by width of the: Ultra High Performance, Enterprise Transactional, Value and Capacity storage options.
+* Storage Class - is how the storage is accessed and used, where:
+  * Platform Native = is managed by the hypervisor / platform (examples are a virtual disk volume from which a VNF  boots and can write back to, the storage interface that is exposed by the container runtime), this storage is typically not shared across running VNF / CNF instances;
+  * Shared Storage = is storage this accessed through a file systems interface (examples are network based storage such as CIFS or NFS) where the storage volumes can be accessed and shared by multiple VNF / CNF instances;
+  * Object Storage = is storage that is accessed via API interaces (the most common example being HTTP restful services API) whichs support get/put of structured objects; and
+  * Archival = is storage that is targetted for provision of long term storage for purpose of disaster recovery, meeting legal requirements or other historical recording where the storage mechanism may go through multiple stages before landing at rest.
+
+The storage model provides a relatively simple way for the storage consumer to specify / select their storage needs. This is shown in the following table which highlights key attributes and features of the storage classes and "epic use cases" for common usage patterns.
 
 | Storage Class | Consumption Model | Performance & Capacity | Cost | Infrastructure Strategy | Use Case |
 |---|---|---|---|---|---|
-| Platform Native | Managed by the VIM / Hypervisor and attached as part of VNF/CNF start up<br />The volumes are not shareable across VNF/CNF instances | Ultra High Performance & Very High Performance<br />Capacity: 10GB - 5TB<br />"Tier 1" | $$$ - $$$$ | Always part of VIM deployment<br />Storage directly next to vCPU<br />Can support highest performance use case<br />Always available to support VNF/CNF boot/startup | Boot/Start VNF/CNF<br />Live Migrate Workload Across VIM |
```

Comment:
In contrast to the definition of "platform native"above, this is very close to the definition of block storage. However, platform native storage volumes provided by a CaaS (persistent volume in K8s) can be shareable as it abstracts aways the underlying type (block or NFS).

Reply:
Hi @gkunz ,

when I was looking at Storage for NVFI at a Telco, it became apparent that there was need to be a way to describe how the storage as consumed via the OS / Application (ie "block", "file" , "object") vs how it was managed by the hypervisor/platform.  This resulted in the introduction of the concept of "hypervisor attached" (in context of OpenStack). Then when Kubernetes was added to solution (with its "Container Persistent Storage") which again is managed via deployment Descriptor  (and analogous to the storage descriptor provided via OpenStack), we had to introduce a new term. The reason was that for Containers the "hypervisor" layer may not exist (in classic sense) so we introduced the concept of "Platform Native". In this case the "Platform" could be OpenStack with hypervisor or Kubernetes without.
Sitting below this within the "Platform" there could be multiple underlying storage technologies used including: In Chassis/DAS, iSCSI or Software Defined Storage such as ScaleIO or Ceph.

Similarly in the VNF running on top, the storage can be provided by the hypervisor / container platform or explicitly mounted via some network exposed storage. Again this could be made available as "block" (via iSCSI say) or "file" (via NFS / CIFS) or "object" (via HTTP) or for something life Ceph as "block" via OS RADOS driver...

In general only OS layer typically talks to "block" and nearly all other cases use either "file" or "object" storage mechanisms..

So when looking at storage need to be clear if we are talking at: OS Layer, Hypervisor/Container Runtime or Application / VNF layers.

Reply:
Hi @jbhartley,
thank you for the explanation. First of all, I don't intent to make this a philosophical discussion on top of this particular PR - therby blocking it getting merged. We can move this discussion to some other place.

My previous comment may also not have been overly clear. What I meant was that "platform native" storage (i.e., storage defined in a platform native descriptor) can also be sharable - hence colliding a bit with the term of "shared storage" and the statement in the table above of "The volumes are not shareable across VNF/CNF instances". The latter could be solved, in alignment with the defintion of "platform native" above, by just stating that the volumes can "_typically_ not be shared" (see line 554 of this file). Maybe that is actually the only fix for this PR and this section I'd like to propose.

In a wider context, outside the scope of this PR: I like the definition of "Platform Native" but I am wondering if it is consistent with "Shared and Object Storage". Wouldn't a more consistent classification be something like:
- platform managed: provided (transparently) by the platform to the application (via a platform native descriptor) and could be shared or non-shared, and
- application managed: the application has to deal with mounting storage (e.g. NFS, iSCSI) or using an API (object), again, those could be shared or non-shared

Reply:
Deferring this discussion to after the Kali release.